src/System.Text.JsonLab.Serialization/System/Text/Json/Serialization/JsonConverter.FromJson.cs

```diff
@@ -16,11 +16,82 @@ public static partial class JsonConverter
 
         private static readonly JsonConverterSettings s_DefaultSettings = new JsonConverterSettings();
 
+        private static void FromJson(
+            ref JsonReaderState readerState,
+            Type returnType,
+            bool isFinalBlock,
+            byte[] buffer,
```

Comment:
nit: Pass in ReadOnlySpan slice instead of array and int?

Reply:
This is called from an async method so it can't pass in ref structs.